wain-exec/src/memory.rs

```diff
@@ -102,13 +102,13 @@ impl Memory {
     pub fn grow(&mut self, num_pages: u32) -> i32 {
         // https://webassembly.github.io/spec/core/exec/instructions.html#exec-memory-grow
         let prev = self.size();
-        let next = prev + num_pages;
+        let next = prev as usize + num_pages as usize;
```

Comment:
修正ありがとうございます．確かに `prev + num_pages` がオーバーフローする可能性がありますね．この直し方でも間違いではないと思うのですが，`usize` で扱わないといけないサイズのページは後で `next_len > MAX_MEMORY_ BYTES` でどうせ弾かれるので，`usize` に変換するよりも `overflowing_add` を使って `u32` のまま扱うで良いかなと思います

```suggestion
        let next, overflow = prev.overflowing_add(num_pages);
        if overflow {
            return -1;
        }
```